### DISCUSSION TITLE
Function declaration shorthand

### DISCUSSION THREAD

**Discussion by orenelbaum** [DanielXMoore/Civet#166] [Ideas]
I'm proposing a new syntax for declaring functions:
```js
f(x: number) = y

// multiline
f(x: number) =
  y := x + 1
  y
```
Which would compile either to a normal function declaration or to an arrow function.
This syntax could also be considered:
```js
f x: number, y: number := x + y
```

**Comment by edemaine** [comment]
Nice, this is like Haskell, though Haskell also allows multiple equations for the same function (overloading).

I wonder whether `f(x: number) := y` with the colon would be more consistent, as this is a declaration? Not sure.
